### DISCUSSION TITLE
How to pull analytics ID?

### DISCUSSION THREAD

**Discussion by kanglicheng** [vercel/community#2866] [Help] [answered]
### Summary

According to this page https://vercel.com/docs/concepts/speed-insights/api, one can access the analytics id via 
process.env.VERCEL_ANALYTICS_ID. 

But in a react app built with CRA the id has to be prefaced with REACT_APP, i.e. REACT_APP_VERCEL_ANALYTICS_ID. I guess this means we need to manually create an environment variable via the vercel dashboard? Is this the recommended way? If this is the case, I need to somehow see the value of VERCEL_ANALYTICS_ID and the docs are not clear on how to do that. 

### Steps to Reproduce

```
const vitalsUrl = 'https://vitals.vercel-analytics.com/v1/vitals';

function getConnectionSpeed() {
  return 'connection' in navigator &&
    navigator['connection'] &&
    'effectiveType' in navigator['connection']
    ? navigator['connection']['effectiveType']
    : '';
}

export function sendToVercelAnalytics(metric) {
  const analyticsId = process.env.REACT_APP_VERCEL_ANALYTICS_ID || process.env.VERCEL_ANALYTICS_ID;
  console.log(analyticsID) // this is undefined
  if (!analyticsId) {
    return;
  }

  const body = {
    dsn: analyticsId,
    id: metric.id,
    page: window.location.pathname,
    href: window.location.href,
    event_name: metric.name,
    value: metric.value.toString(),
    speed: getConnectionSpeed(),
  };

  const blob = new Blob([new URLSearchParams(body).toString()], {
    // This content type is necessary for `sendBeacon`
    type: 'application/x-www-form-urlencoded',
  });
  if (navigator.sendBeacon) {
    navigator.sendBeacon(vitalsUrl, blob);
  } else
    fetch(vitalsUrl, {
      body: blob,
      method: 'POST',
      credentials: 'omit',
      keepalive: true,
    });
}
```

**Comment by amyegan** [comment, marked as answer]
Hi @kanglicheng! I moved your post from the CLI repo over to our support community.

`VERCEL_ANALYTICS_ID` is inlined during the build process along with a version using the appropriate client side prefix. It should not be added to your project's environment variables manually. 

You can find an example of `REACT_APP_VERCEL_ANALYTICS_ID` being used in the [Create React App starter](https://github.com/vercel/vercel/blob/main/examples/create-react-app/src/vitals.js).
